File: Blind75/104. Maximum Depth of Binary Tree/104. Maximum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int res = 0;
    int maxDepth(TreeNode* root) {
        if(root==NULL) return 0;
        DFS(root,1);
        return res;
    }

    void DFS(TreeNode* node, int depth){
        if(node==NULL) return;

        res = max(res, depth);
        DFS(node->left, depth+1);
        DFS(node->right, depth+1);
    }
};

File: Blind75/104. Maximum Depth of Binary Tree/104. Maximum Depth of Binary Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int MAX = 0;
    public int MaxDepth(TreeNode root) {
       DFS(root,1);
       return MAX;
    }

    public void DFS(TreeNode node, int depth){
        if(node==null) return;
        MAX = Math.Max(MAX, depth);
        DFS(node.left,depth+1);
        DFS(node.right,depth+1);
    }
}

